openedx/core/djangoapps/schedules/resolvers.py

```diff
@@ -1,136 +1,313 @@
 import datetime
```

Comment:
Consider moving each message type's resolver into its own file.  It could make the code more navigable.
Same suggestion for the code in tasks.py.

Reply:
Yeah, I'll take a look at that.

Reply:
For the time being, I rearranged some of the code in tasks.py and further DRYed the individual tasks.  We can consider this in a subsequent PR.